### PR TITLE
User search

### LINKED ISSUE #253: User search
expose a user search box like aurora
`/v1/users/?search[]=``

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Adds the ability to search for a user based on ID, Email, or mobile number.

#### How should this be manually tested?

Pull down branch, go to users page or contestants page and search for a user based on Email, ID, or number and you should find them.

#### What are the relevant tickets?
Fixes #253 

@weerd @angaither @deadlybutter

### CODE REVIEW

File: app/Http/Controllers/UsersController.php [hunk 1]
```diff
@@ -169,4 +170,24 @@ public function contestantsIndex()
...
+
+        // Redirect empty queries to the user index.
+        if ($query === '') {
+            return redirect()->route('users.index');
+        }
+
+        // Attempt to fetch all users.
+        $users = $this->manager->search($query);
+
+        return $users ? redirect()->route('users.show', [$users->first()->id]) : redirect()->route('users.index')->with('status', 'No user found!');
```
Comment (deadlybutter) [inline app/Http/Controllers/UsersController.php]: is there any case we might want to search for multiple users?

Comment (deadlybutter) [inline app/Http/Controllers/UsersController.php]: hmm probably not. if we need to see a group of users we have other pages for that.

Comment (sbsmith86) [inline app/Http/Controllers/UsersController.php]: i wasn't sure about that, but it doesn't seem that useful, we could also look into some like filtering options on the contestants page.

File: app/Services/Manager.php [hunk 2]
```diff
@@ -528,4 +528,26 @@ public function sendEmail($user, $contest, $params)
...
+     */
+    public function search($query, $page = null)
+    {
+        // Attempt to fetch all users.
+        $users = $this->northstar->getAllUsers([
+            'search' => [
+                '_id' => $query,
+                'drupal_id' => $query,
+                'email' => $query,
+                'mobile' => $query,
```
Comment (deadlybutter) [inline app/Services/Manager.php]: I'm surprised Northstar doesn't freak out if you supply an email address for a mobile number?

**Comment by angaither** [comment]
pics or it didn't happen

File: app/Services/Manager.php [hunk 3]
```diff
@@ -528,4 +528,26 @@ public function sendEmail($user, $contest, $params)
             return null;
         }
     }
+
+    /**
+     * Search for users in Northstar.
+     *
+     * @param  string $query
```
Comment (deadlybutter) [inline app/Services/Manager.php]: missing page param

**Comment by sbsmith86** [comment]
lol @angaither 

user-search

**Comment by deadlybutter** [comment]
other than the little comment 👍

File: app/Services/Manager.php (hunk 3 above)
Comment (sbsmith86) [inline app/Services/Manager.php]: ah yup

**Comment by angaither** [comment]
@sbsmith86 does it work for contestants too?

**Comment by sbsmith86** [comment]
@angaither it's the same search functionality just on different pages. 

contestant-search

**Comment by angaither** [comment]
💯💯💯

**Comment by angaither** [comment]
Oh wait though. Don't we want to be able to have one box that's searches either?

**Comment by sbsmith86** [comment]
It does! It searches users in northstar and shows them. I just put the search box on two different pages.

**Comment by angaither** [comment]
oooooookie :shipit:

**Pull request merged**